ql/experimental/credit/homogeneouspooldef.hpp

```diff
@@ -142,7 +142,7 @@ namespace QuantLib {
             detachAmount_);
             //notional_);
         std::vector<Real> mkft(1, min_ + delta_ /2.);
-        for (Size i = 0; i < nSteps_; i++) {
+        for (Real i = 0; i < nSteps_; i++) {
```

Comment:
Generally it's better to use integral types for the loop variable in order to avoid any floating point comparison issues. How about you keep the loop variable as `Size` but change the comparison to `static_cast<Size>(nSteps_)`?

Reply:
I think nSteps_ should have type Size from the beginning. We can then cast to Real when needed, e.g. when computing delta.

Reply:
Agree - the `nSteps_` is already fishy in the original code - we had the choice to either make `i` `Real` or change `nSteps_`. Let's make them both `Size`, that's definitely better.

Reply:
Made this a `Size` now and reverted the iterator variable.